fe/fe-core/src/main/java/org/apache/doris/nereids/types/DataType.java

```diff
@@ -49,6 +54,26 @@ public abstract class DataType implements AbstractDataType {
             .put(FloatType.class, () -> DoubleType.INSTANCE)
             .build();
 
+    /**
+     * create a specific Literal for a given dataType
+     */
+    public static Literal promoteNumberLiteral(Object value, DataType dataType) {
+        if (! (value instanceof Number)) {
+            return null;
+        }
+
+        if (dataType.equals(SmallIntType.INSTANCE)) {
+            return new SmallIntLiteral(((Number) value).shortValue());
```

Comment:
should handle overflow?

Reply:
offline discussed. done
we need another pr to do down-cast